.github/actions/check_version/check_versions_are_all_identical.sh

```diff
@@ -0,0 +1,29 @@
+# Leka - LekaOS
+# Copyright 2023 APF France handicap
+# SPDX-License-Identifier: Apache-2.0
+
+shopt -s xpg_echo
+
+ARE_ALL_SAME=True
+
+if [ $OS_VERSION_FILE_ARE_SAME != "True" ]; then
+	ARE_ALL_SAME=False
+fi
+
+if [ $OS_VERSION_ARE_SAME != "True" ]; then
+	ARE_ALL_SAME=False
+fi
+
+if [ $FIRMWARE_VERSION_FILE_ARE_SAME != "True" ]; then
+	ARE_ALL_SAME=False
+fi
+
+if [ $FIRMWARE_VERSION_ARE_SAME != "True" ]; then
+	ARE_ALL_SAME=False
+fi
+
+if [ $ARE_ALL_SAME != "True" ]; then
+	exit 1
+fi
```

Comment:
as you actually don't care which one is wrong (it will be shown in the comment in Github), you would simplify like this:

```suggestion
if [ $OS_VERSION_FILE_ARE_SAME != "True" ]; then
	exit 1
fi

if [ $OS_VERSION_ARE_SAME != "True" ]; then
	exit 1
fi

if [ $FIRMWARE_VERSION_FILE_ARE_SAME != "True" ]; then
	exit 1
fi

if [ $FIRMWARE_VERSION_ARE_SAME != "True" ]; then
	exit 1
fi
```